projectiles/projectile_effect.gd

```diff
@@ -0,0 +1,75 @@
+class_name ProjectileEffect
+extends Node
+# A ProjectileEffect is a set of post-processors meant to be run on every instance of a Projectile.
+# This by itself does nothing but is meant to be extended by various types of projectile effect.
+# Each effect will compound on the others. Each function will be called by the Projectile scene,
+#    in order of Effect Priority
+# TODO:: Effect Priority as an enum? (to discourage "priority: 9999999")
+
+# Each modifier is a combination of a double value and a modification type.
```

Comment:
❓ Non-blocking: Do we want to only do additive? In that it also allows multiplicative i.e. multiple additive nodes can be applied to get multiplicative effects.

Reply:
Not sure I follow.

Additive can be achieved if the projectile effect is the only one modifying that attribute of the projectile, but would not work if we wanted two different effects to compound on eachother.

Ex: We have a "speed multiplier" effect that makes the projectile go 10x faster, and we have a "rate stack effect" pickup, for each one you pick up you get +5 to fire rate and projectile speed.

So you could have 3 additive stacks, followed by a multiplier

Reply:
Yep fair point 👍